nncf/binarization/extensions.py

```diff
@@ -36,15 +34,34 @@
     os.path.join(BASE_EXT_DIR, "cuda/functions_cuda_impl.cu")
 ]
 
-BinarizedFunctionsCPU = load(
-    'binarized_functions_cpu', CPU_EXT_SRC_LIST, extra_include_paths=EXT_INCLUDE_DIRS,
-    verbose=False
-)
+
+@EXTENSIONS.register()
+class BinarizedFunctionsCPULoader:
```

Comment:
You should make this and other related classes implement a specific abstract class interface instead of just creating the same number of methods with the same signature.

BTW, if you fancy, you may try using metaclasses or `__init_subclass__` to handle registration automatically, but this is not a strict requirement - `.register()` already does the trick.

Reply:
Implemened; please take a look at the updated commit